includes/server-timing/load.php

```diff
@@ -156,21 +175,17 @@ function perflab_register_server_timing_setting() {
  *
  * @since 2.6.0
  *
- * @param mixed $value Server-Timing setting value.
- * @return array Sanitized Server-Timing setting value.
+ * @param array|mixed $value Server-Timing setting value.
+ * @return array{benchmarking_actions: string[], benchmarking_filters: string[], output_buffering: bool} Sanitized Server-Timing setting value.
  */
-function perflab_sanitize_server_timing_setting( $value ) {
-	static $allowed_keys = array(
-		'benchmarking_actions' => true,
-		'benchmarking_filters' => true,
-		'output_buffering'     => true,
-	);
-
+function perflab_sanitize_server_timing_setting( $value ): array {
 	if ( ! is_array( $value ) ) {
-		return array();
+		$value = array();
 	}
-
-	$value = array_intersect_key( $value, $allowed_keys );
+	$value = wp_array_slice_assoc(
+		array_merge( perflab_get_server_timing_setting_default_value(), $value ),
+		array_keys( perflab_get_server_timing_setting_default_value() )
+	);
```

Comment:
This ensures the default value is always stored, while also being the default option in case it is not stored.